packages/accounts-controller/src/utils.ts

```diff
@@ -79,3 +79,15 @@ export function isNormalKeyringType(keyringType: KeyringTypes): boolean {
   // adapted later on if we have new kind of keyrings!
   return keyringType !== KeyringTypes.snap;
 }
+
+/**
+ * Creates a deep clone of the given object.
+ * This is to get around error `Type instantiation is excessively deep and possibly infinite.`
+ *
+ * @template O,T - The type of the object being cloned.
+ * @param obj - The object to be cloned.
+ * @returns The deep clone of the object.
+ */
+export function deepCloneDraft<I, T>(obj: I): T {
+  return JSON.parse(JSON.stringify(obj)) as T;
+}
```

Comment:
Not sure about this solution. I can foresee some problems if the data type has Map, Set, RegExp objects. And also the parse and stringify operations are quite expensive in terms of performance for large objects which is what the error suggests in this case. Lastly, stringify is a bit unpredictable in some cases.

I'd strongly suggest to use another solution for this case.

Reply:
- https://developer.mozilla.org/en-US/docs/Glossary/Deep_copy
- https://developer.mozilla.org/en-US/docs/Web/API/structuredClone
- https://saturncloud.io/blog/what-is-the-most-efficient-way-to-deep-clone-an-object-in-javascript/#method-3-recursive-function

Looking at some benchmarks, recursive deep cloning looks like to be the most effective one.

Reply:
Looks like on the Snap side, they are using this: https://www.npmjs.com/package/rfdc

So we're gonna use their `deepClone` (from `snaps-utils`)

Reply:
structuredClone was not available in node 16